posthog/caching/login_device_cache.py

```diff
@@ -0,0 +1,24 @@
+import hashlib
+from posthog.redis import get_client
+
+
+def check_and_cache_login_device(user_id: int, ip_address: str, short_user_agent: str) -> bool:
```

Comment:
We should also add tests for this if you're able to figure out the redis tests, should be lots of examples.